### DISCUSSION TITLE
Source Google Analytics: add multichannel funnels API

### DISCUSSION THREAD

**Discussion by raismartup** [airbytehq/airbyte#36091] [New Connector Request]
## Tell us about the new connector you’d like to have
* Which source and which destination?
source google analytics api 
**https://developers.google.com/analytics/devguides/reporting/mcf/v3/reference**

## Describe the context around this new connector
* Why do you need this integration? How does your team intend to use the data? This helps us understand the use case.
necessary to integrate when we want ETL on multichannel funnel data 
* How often do you want to run syncs?
Daily
* If this is an API source connector, which entities/endpoints do you need supported?
authentication: https://www.googleapis.com/auth/analytics.readonly

referencs: https://developers.google.com/analytics/devguides/reporting/mcf/dimsmets

* If the connector is for a paid service, can we name you as a mutual user when we subscribe for an account? Which company should we name? 
Not paid

## Describe the alternative you are considering or using
What are you considering doing if you don’t have this integration through Airbyte?

Make an ETL in local code, specifically in python.

## Are you willing to submit a PR?

I don't know Java and therefore it would take me a lot of time. If someone who can develop it would be appreciated and anything I can do to help, count me in.

**Comment by natalyjazzviolin** [comment]
Hi @raismartup , we currently already support the Google Analytics API:
https://docs.airbyte.com/integrations/sources/google-analytics-v4/

Could you please elaborate more on what features specifically you'd like to be added to the connector? Thank you!

**Comment by raismartup** [comment]
Hello again.
Thanks for your reply.

Correct , Airbyte already has Google Analytics API support available but it is focused on acquisition data, not multichannel funnels.
**https://developers.google.com/analytics/devguides/reporting/core/v4/quickstart/service-java**
**https://ga-dev-tools.web.app/dimensions-metrics-explorer/**

usually you can recognise the dimensions and metrics returned by this API with the name "ga:...", for example, ga:date, ga:adCost, ga:adClicks, etc.

But the API I am referring to is the multichannel funnels API.
**https://developers.google.com/analytics/devguides/reporting/mcf/v3/reference**

You can see that the dimensions and metrics are different, it is usually recognised by "mcf:..." as for example, mcf:SourceMedium, mcf:LastInteractiveValue, etc.

I was testing to see if Airbyte could bring these metrics "mcf:.." and I had no success because it does not recognise them and it must be because it is a different API than the usual GA4 one

If you need more information or an example of the call difference between both APIS, ask me for it.

Thank so much.

**Comment by natalyjazzviolin** [comment]
@raismartup thank you so much for the write up and sorry I missed that detail!

**Comment by leo-schick** [comment]
Hello @raismartup,

I think this will not work. The Multi-Channel-Funnel API v3 does not support GA4 properties, see [here](https://developers.google.com/analytics/devguides/reporting/mcf/v3) and since 01.07.2023, all data in Google Analytisc Universal is frozen.

Is there another API you are aware of which could be used to get the Multi-Channel-Funnel data in GA4?

p.s. if you still want to download the frozen MCF data, you could maybe take a look at my package [mara-google-analytics-downloader](https://github.com/leo-schick/mara-google-analytics-downloader) which gives you an CLI tool to download the data manually. Check out `mara-google-analytics-downloader --help` after installing the python package locally.

[a comment by an automated account is not shown]
